### PR TITLE
fix deprecated asyncCheck

### LINKED ISSUE #2458: Fix deprecated asyncCheck usage warnings
- [x] nimbus-eth2/tests/test_peer_pool.nim(458, 5) Warning: Raises Defect on future failure, fix your code and use asyncSpawn!; asyncCheck is deprecated [Deprecated]

- [x] nimbus-eth2/ncli/inspector.nim(741, 3) Warning: Raises Defect on future failure, fix your code and use asyncSpawn!; asyncCheck is deprecated [Deprecated]

- [x] nimbus-eth2/ncli/inspector.nim(751, 7) Warning: Raises Defect on future failure, fix your code and use asyncSpawn!; asyncCheck is deprecated [Deprecated]

- [x] nimbus-eth2/ncli/inspector.nim(754, 5) Warning: Raises Defect on future failure, fix your code and use asyncSpawn!; asyncCheck is deprecated [Deprecated]

- [x] nimbus-eth2/beacon_chain/nimbus_validator_client.nim(276, 5) Warning: Raises Defect on future failure, fix your code and use asyncSpawn!; asyncCheck is deprecated [Deprecated]

- [x] nimbus-eth2/beacon_chain/nimbus_validator_client.nim(340, 7) Warning: Raises Defect on future failure, fix your code and use asyncSpawn!; asyncCheck is deprecated [Deprecated]

### PR DESCRIPTION
**Pull request by bernardoaraujor**
Resolve https://github.com/status-im/nimbus-eth2/issues/2458

### CODE REVIEW

**Comment by arnetheduck** [comment]
Thanks for the PR - normally, we do development work against the `unstable` branch which incidentally just had these issues fixed by way of deprecating the old validator client code.

**Comment by arnetheduck** [comment]
Sorry, just saw that the inspector was fixed here as well - this PR will need a rebase against `unstable`.

**Comment by bernardoaraujor** [comment]
I'll check whether there's any unhandled exceptions that `asyncSpawn` could allow to arise.
I'll report here soon.

File: ncli/inspector.nim [hunk 1]
```diff
@@ -742,11 +744,13 @@ proc run(conf: InspectorConf) {.async.} =
     var proto = bootstrapDiscovery(conf, hostAddress.get(), seckey,
                                    disc5bootnodes, enrFields)
     if not(conf.noDiscovery):
-      asyncCheck discoveryLoop(conf, proto, switch, connectQueue,
+      asyncSpawn(try: discoveryLoop(conf, proto, switch, connectQueue,
                                pubsubPeers)
+      except CatchableError as exc: raise exc)
```
Comment (bernardoaraujor) [inline ncli/inspector.nim:749]: @arnetheduck would this be an adequate way to have `asyncSpawn` handle exceptions?

`discoveryLoop` has only a single `except CatchableError` declaration.

File: ncli/inspector.nim [hunk 2]
```diff
@@ -742,11 +744,13 @@ proc run(conf: InspectorConf) {.async.} =
...
 
-    asyncCheck resolveLoop(conf, proto, switch, resolveQueue,
+    asyncSpawn(try: resolveLoop(conf, proto, switch, resolveQueue,
                            pubsubPeers)
+    except CatchableError as exc: raise exc)
```
Comment (bernardoaraujor) [inline ncli/inspector.nim:753]: ditto

File: ncli/inspector.nim (hunk 1 above)
Comment (arnetheduck) [reply ncli/inspector.nim:749]: not quite - any exceptions need to be handled _inside_ `discoveryLoop` - `async` is a bit confusing this way, in that exceptions are "deferred" until `await` (or `Future.read`) is called - they never directly emanate from  an `async` function

**Comment by tersec** [comment]
ncli/inspector.nim has since been removed entirely, rendering the changes here moot.

**Comment by bernardoaraujor** [comment]
ok, closing.

**Pull request closed without merging**